util/system/path.go

```diff
@@ -16,3 +22,80 @@ func DefaultPathEnv(os string) string {
 	}
 	return DefaultPathEnvUnix
 }
+
+// NormalizeWorkdir will return a normalized version of the new workdir, given
+// the currently configured workdir and the desired new workdir. When setting a
+// new relative workdir, it will be joined to the previous workdir or default to
+// the root folder.
+// On Windows we remove the drive letter and convert the path delimiter to "\".
+// Paths that begin with os.PathSeparator are considered absolute even on Windows.
+func NormalizeWorkdir(current, wd string) (string, error) {
+	if current == "" {
+		current = "/"
+	}
+
+	var err error
+	current, err = CheckSystemDriveAndRemoveDriveLetter(current)
+	if err != nil {
+		return "", errors.Wrap(err, "removing drive letter")
+	}
+
+	current = filepath.FromSlash(current)
+	if !IsAbs(current) {
+		// Convert to absolute paths.
+		//
+		// On Windows:
+		// Paths that start with a / or \ are absolute paths relative to the current drive
```

Comment:
Paths starting with a / or \ can also be UNC paths especially in Windows if it is \\ or //

Reply:
Indeed. Currently, if we use ```\\.\C$\test``` as a ```WORKDIR``` in Docker, we end up with a container that has it's CWD set to:

```cmd
Microsoft Windows [Version 10.0.20348.1249]
(c) Microsoft Corporation. All rights reserved.

C:\C$\workdir6>
```

I need to check if HCS will allow a UNC path as a ```WORKDIR```. If not, we should probably err out here.

Reply:
UNC paths are not supported as a CWD for containers. I changed the code to error out if we get a UNC path.